framework/src/problems/FEProblemBase.C

```diff
@@ -4304,216 +4304,218 @@ FEProblemBase::computeUserObjectsInternal(const ExecFlagType & type,
+          create_and_run_mortar_functors(false);
+          if (_displaced_problem)
+            create_and_run_mortar_functors(true);
+        }
+        for (auto obj : mortar)
+          obj->finalize();
       }
-      for (auto obj : mortar)
-        obj->finalize();
-    }
 
-    // Execute threaded general user objects
-    for (auto obj : tgobjs)
-      obj->initialize();
-    std::vector<GeneralUserObject *> tguos_zero;
-    query.clone()
-        .condition<AttribThread>(0)
-        .condition<AttribInterfaces>(Interfaces::ThreadedGeneralUserObject)
-        .queryInto(tguos_zero);
-    for (auto obj : tguos_zero)
-    {
-      std::vector<GeneralUserObject *> tguos;
-      auto q = query.clone()
-                   .condition<AttribName>(obj->name())
-                   .condition<AttribInterfaces>(Interfaces::ThreadedGeneralUserObject);
-      q.queryInto(tguos);
-
-      ComputeThreadedGeneralUserObjectsThread ctguot(*this);
-      Threads::parallel_reduce(GeneralUserObjectRange(tguos.begin(), tguos.end()), ctguot);
-      joinAndFinalize(q);
+      // Execute threaded general user objects
+      for (auto obj : tgobjs)
+        obj->initialize();
+      std::vector<GeneralUserObject *> tguos_zero;
+      query.clone()
+          .condition<AttribThread>(0)
+          .condition<AttribInterfaces>(Interfaces::ThreadedGeneralUserObject)
+          .queryInto(tguos_zero);
+      for (auto obj : tguos_zero)
+      {
+        std::vector<GeneralUserObject *> tguos;
+        auto q = query.clone()
+                     .condition<AttribName>(obj->name())
+                     .condition<AttribInterfaces>(Interfaces::ThreadedGeneralUserObject);
+        q.queryInto(tguos);
+
+        ComputeThreadedGeneralUserObjectsThread ctguot(*this);
+        Threads::parallel_reduce(GeneralUserObjectRange(tguos.begin(), tguos.end()), ctguot);
+        joinAndFinalize(q);
+      }
+
+      // Execute general user objects
+      joinAndFinalize(query.clone().condition<AttribInterfaces>(Interfaces::GeneralUserObject),
+                      true);
     }
+  }
 
-    // Execute general user objects
-    joinAndFinalize(query.clone().condition<AttribInterfaces>(Interfaces::GeneralUserObject), true);
+  catch (libMesh::LogicError & e)
+  {
+    setException("The following libMesh::LogicError was raised during UserObject evaluation in "
+                 "FEProblemBase:\n" +
+                 std::string(e.what()));
+  }
+  catch (MooseException & e)
+  {
+    setException(
+        "The following MooseException was raised during UserObject evaluation in FEProblemBase:\n" +
+        std::string(e.what()));
+  }
+  catch (MetaPhysicL::LogicError & e)
+  {
+    moose::translateMetaPhysicLError(e);
+  }
+  try
+  {
+    // Propagate the exception to all processes if we had one
+    checkExceptionAndStopSolve();
+  }
+  catch (MooseException &)
+  {
+    // Just end now. We've inserted our NaN into the residual vector
```

Comment:
Paired with my comment above, can we `mooseAssert(type == EXEC_LINEAR, "We should only be continuing if this is a linear/residual evaluation and consequently we know we've inserted a NaN in the residual vector");` or some message like that right here? Then we are confident in this comment